src/shared/bus-polkit.c

```diff
@@ -530,11 +530,13 @@ int bus_verify_polkit_async_full(
         }
 #endif
 
-        r = sd_bus_query_sender_privilege(call, -1);
-        if (r < 0)
-                return r;
-        if (r > 0)
-                return 1;
+        if (!FLAGS_SET(flags, BUS_POLKIT_FLAGS_SKIP_UID_CHECK)) {
+                r = sd_bus_query_sender_privilege(call, -1);
+                if (r < 0)
+                        return r;
+                if (r > 0)
+                        return 1;
+        }
```

Comment:
this flag must be plumbed though for the varlink case too

also, a patch like this is already part of #26826